js/iframe-coordinator/libs/workers/worker-manager.ts

```diff
@@ -0,0 +1,288 @@
+// import [* as] SpawnWorker from "../workers/SpawnWorker.worker.ts";
+/* Exporting as a class gets by the typescript check and loads the file, however, the constructor
+isn't called.  Also, the umd version of the module can't find the default since the worker-loader
+seems to export as a constructor function anyway.  So, the raw text version seems to be the best
+approach at this point*/
+import {
+  WORKER_MESSAGING_PROTOCOL_NAME,
+  WorkerLifecycleEvents,
+  WorkerToHostMessageTypes
+} from './constants';
+import * as SpawnWorker from './spawn-worker.worker.ts';
+
+export const WORKER_MESSAGE_EVENT_TYPE: string = 'workermessage';
+
+const DEFAULT_ERROR_WINDOW_COUNT_THRESHOLD = 10;
+const DEFAULT_ERROR_WINDOW_MILLIS = 30000;
+
+/*
+ * TODO for this feature:
+
+ * get bi-directional comms working
+ * Finalize backgroundClient api
+   * Check on impl interface idea and add NavigationRequester interface
+   * should navRequest be url based?
+   * Will docs get pulled in correctly? pull them from client.ts
+ * Message validation
+  * Add checks to ensure lifecycle is not updated to prior state?
+  * Ensure payload is valid for message types
+  * Better checks to ensure all enums and cases are handled?
+  * stronger types of messages between this manager and the workers?
+ * docs
+ *   dont forget that this is sorta dangerous with indexeddb
+ *
+ * Is the import of the worker good enough?
+  * Decide on class or text of the worker
+ * Make sure es6 target is ok
+ * Make sure it works in IE
+ *  check blob and fallback
+ * Improve worker demo for the real, useful demo.
+ *  use subscribe?
+ *  allow dynamic addition?
+ * Tests
+ *  tests for error rate
+ * better logging (these are important)
+ * support lenient shutdown?
+ *  notification on shutdown?
+ *  soft stop
+ *  allow restart?
+ * support client stop request?
+ *   worker never can close now as i have attached listeners
+ *   but, you can't re-add now either via registerclients
+ */
+
+enum WorkerPhase {
```

Comment:
@stormojm ... here's the worker lifecycle I mentioned in the other PR.  I would be 100% on board with elevating this to a ClientLifecycle so we have consistent phases and events to advance the client through.  It could also be extended if clients have different phases, etc.s